### ISSUE TITLE
Suggestion RPC Fix

### ISSUE THREAD

**Issue by Woodi-dev** [NotHunter101/ExtraRolesAmongUs#53]
Hey first of all great Mod.
I saw you took some code snippets from my Sheriff mod Code. However, this was my first mod and there are some bugs included.

Please do not use
StartRpcImmediately with SendOption None for Network messages since the message may go lost.

Use 			
MessageWriter writer = FMLLKEACGIO.Instance.StartRpc(NET ID, RPC CALL, Hazel.SendOption.Reliable);
writer.EndMessage();

Regards,
Woodi

**Comment by NotHunter101** [comment]
Whoops, that may actually be why a lot of problems occur with applying roles. Thanks a lot, I'll fix this when I can. I loved your Sheriff mod, by the way!

**Comment by Woodi-dev** [comment]
Thank you. No problem :)

**Comment by rpet91** [comment]
Would love to see the Love Couple mod merged with this one. :D

**Comment by psibean** [comment]
Does this mean ALL of the `StartRpcImmediately` calls need to be updated?

**Comment by Woodi-dev** [comment]
> Does this mean ALL of the `StartRpcImmediately` calls need to be changed?
No. You can use this method. It is faster than startrpc. But more important is the SendOption Reliable. Selecting the player role is some information which everybody has to get. If you wanna send other information like player position or smth you can use SendOption None.

**Comment by Popeye4242** [comment]
https://github.com/NuclearPowered/Reactor/blob/master/Reactor.Example/ExampleRpc.cs

All calls should be changed to Reactor CustomRpcs. Using them means this mod is compatible with other mods because the custom RPC id's don't overlap. Any mod should use this implementation.

**Comment by Woodi-dev** [comment]
> https://github.com/NuclearPowered/Reactor/blob/master/Reactor.Example/ExampleRpc.cs
> 
> All calls should be changed to Reactor CustomRpcs. Using them means this mod is compatible with other mods because the custom RPC id's don't overlap. Any mod should use this implementation.
> 
> ```
>  Rpc<ExampleRpc>.Instance.Send(new ExampleRpc.Data("Cześć :)"));
> Rpc<ExampleRpc>.Instance.SendTo(AmongUsClient.Instance.HostId, new ExampleRpc.Data("host :O"));
> ```

Of course could change that easily. RPC calls like in my Sheriff mod were made before Reactor. The problem is that changing RPC net code makes automatically new mod version incompatible to older versions

**Comment by NotHunter101** [comment]
Still thinking about whether I want to actually switch to the Reactor RPC implementation because of the reason above. Either way, I am going to be fixing this asap, I have just been really busy with mod commissions.

**Comment by Popeye4242** [comment]
Fixed with #90

**Comment by Netskeh** [comment]
might be a little late to chime in, here are my 2 cents:
since there is a mod breaking game update anyway (as far as I can tell, I did not play with the mod since) - one could use the opportunity to do this kind of change as well
